packit/config/package_config.py

```diff
@@ -601,3 +624,37 @@ def get_specfile_path_from_repo(
         logger.debug(f"No spec file found in {project.full_repo_name!r}")
         return None
     return spec_files[0]
+
+
+def merge_configs(base: dict, override: dict) -> dict:
+    """
+    Deep merge override config into base config.
+    Override values take precedence.
+    """
+    from copy import deepcopy
+
+    result = deepcopy(base)
+
+    for key, value in override.items():
+        if key in result and isinstance(result[key], dict) and isinstance(value, dict):
+            result[key] = merge_configs(result[key], value)
+        else:
+            result[key] = value
+
+    return result
```

Comment:
This definitely needs to be covered by tests. Not for the sake of testing, but for the sake of defining the expected behavior.